travelpal/lib/travelpal_web/controllers/flight_controller.ex

```diff
@@ -0,0 +1,70 @@
+defmodule TravelpalWeb.FlightController do
+  use TravelpalWeb, :controller
+
+  # @TODO decide if other functions are needed
+  def flight_url, do: "https://api.skypicker.com/flights"
+
+  def get_flights_to_from(conn, %{"origin" => origin, "dest" => dest, "date_from" => date_from, "return_from" => return_from}) do
+    # gets top 5 flights
+    uri = URI.encode(flight_url() <> "?flyFrom=#{origin}&to=#{dest}&date_from=#{date_from}&date_to=#{date_from}"
+        <> "&return_from=#{return_from}&return_to=#{return_from}&partner=picky&partner_market=us&limit=5")
+    
+    # comment out HTTP request for dev purposes and use dummy data instead
+    #res = HTTPoison.get!(uri)
+    #data = Poison.decode!(res.body)
+
+    # use dummy data for dev purposes
+    #flight = data["data"]
+    # @TODO decide if more details are needed
+    flight = dummy_data()
+    |> Enum.map(fn(x) -> Map.take(x, ["price", "mapIdfrom", "mapIdto", "airlines", "duration", "deep_link"]) end)
+    |> Poison.encode!()
+
+    render(conn, "show.json", flight: flight)
+  end
+
+  def dummy_data do
+    [
+      %{
+        "airlines" => ["AS", "UA"],
+        "deep_link" => "https://www.kiwi.com/deep?from=BOS&to=LAX&departure=05-08-2018&return=05-12-2018&flightsId=3571213871549614_0%7C3927455603931439_0&price=320&passengers=1&affilid=picky&lang=en&currency=EUR&booking_token=TjFsU9KOyjpEDqm5dNSrO1ZjuRMVMgZrERrSppD325T6P8+P9w468E+3gYirNKm0xjFEaQ8brFkJXYHcFnHHQMiKg0Dkoaiav5UhUit795jeFFAqEo1EKVZuez5O1ueoor1vahBPhrpq/0WADMhABtY5vdWSberiYk61VFIJMRp4s6ZSzHyNgbNckdTqU7qdfZpRqVJEtW6IRWoeCiw1GPFDI+Hw1yLpRfB1ptE7vlW9tkkPUjw697wUdcL4Yx28aUCVJRBrCA/L3FNex/RY6mOqHthZ8JQWOD6Y0YUjhUwPGmJACX0xuu88UBNWGVbG3f9HTq2CpG5Ia7H1eZs+S+5Zbl9TxM1P/0iyFcvHGI0DEZeHss34gti+hRSv2W+DDTTJeq1SPB6fOVj+rGl20MF+nQmjjpqw86rcq6cG5Fi706sm7iciWAjIhCba7quYWYzIotUE40rtoc7GuSrSZCk3uO4zcygslqkHcKfisWyMazFn4mEAp3zeWxi8yxSNtZoH+CIGLxT4z/9MizxZQB4ScDmb/ogmDP77vENDVNPOYBVb3hQ0a4sqe8h0na2ExjcDaiubr0TK8M0xCv07UwMl7ZzZCpun4o1ca+H1SKu6cb9FgpHUe1mHfiPB6acjuL7K9kr1hdvIpo3aE24koYDcxxkHFAkgQB6O2L/qCUPQYRurE9MdVAOH8gnVaY0O54TPTy4XBAjKuhCRyE4NrnsNxM8UYnbZfH6XXYs4rCI=",
+        "duration" => %{"departure" => 22800, "return" => 19200, "total" => 42000},
+        "mapIdfrom" => "boston",
+        "mapIdto" => "los-angeles",
+        "price" => 320
+      },
+      %{
+        "airlines" => ["UA"],
```

Comment:
Might be obvious but do we have a map to translate these airline encoding to their full names?

Reply:
Yeah I'll have to find a list of all of them and I can just search it to get the name.